### ISSUE TITLE
[REQ] Allow generators to provide a JSON schema to validate their config

### ISSUE THREAD

**Issue by dkarlovi** [OpenAPITools/openapi-generator#7534]
### Is your feature request related to a problem? Please describe.

Generators support quite a few options which you pass via the config. Having that config be validated would allow for better UX since the user would know about typos, etc.

Example
```
.openapi-generator.json: fail: (root): Additional property ianvokerPackage is not allowed
```

An example schema: https://github.com/dkarlovi/openapi-generator/blob/master/generators/php/config/schema.json

## Describe the solution you'd like

Each generator might provide a schema to generate the generator-specific config. It could get merged with the common config which could be provided to all generators.

**Comment by wing328** [comment]
Agreed with better error message to report invalid options but the challenge is one can also pass mustache variables via additional properties and use these in the templates, e.g. `--additional-properties newTag=newValue` means users can use `{{newTag}}` in their customized templates.

**Comment by dkarlovi** [comment]
@wing328 I'm not sure I see your point, how do the additional properties influence validating the config file? :thinking:
